### ISSUE TITLE
Graphical issues; Bad OpenGL Calls? (Stack overflow 1283, POST RENDER)

### ISSUE THREAD

**Issue by VenomiXeD** [micdoodle8/Galacticraft#3810]
Minecraft version: 10.13.4.1614, Forge 1.7.10.
Galacticraft version:
GalacticraftCore-1.7-3.0.12.504.jar
Galacticraft-Planets-1.7-3.0.12.504.jar
MicdoodleCore-1.7-3.0.12.504.jar

Single player (SSP), Multiplayer (SMP), or SSP opened to LAN (LAN)?
SMP, Standard Forge & SSP

Galacticraft add-ons installed:
0

Please provide screenshots / crash report / log / full modlist as needed
![image](https://user-images.githubusercontent.com/32819948/66704528-0548c380-ed1d-11e9-9307-1b1464e3fb4c.png)
![image](https://user-images.githubusercontent.com/32819948/66704557-4b058c00-ed1d-11e9-938b-84b780c52afe.png)

Small testing mod list
![image](https://user-images.githubusercontent.com/32819948/66704585-830ccf00-ed1d-11e9-9feb-ce94952e25bf.png)

This is a severe graphical bugs. The console constantly spam "OpenGL Stack overflow". Bad OpenGL calls? I've also tried removing all mods and just include GalacticCraft and a mod that modifies the player. This also causes water to be stuck in the screen. I've already come in contact with Vampirism's mod developers and it's not on their end. As it works fine without gc and just with the Vampirism mod. However. Vampirism is not the only mod. It also includes "howling moon" that modifies the player to become a wolf.

**Comment by radfast** [comment]
Galacticraft 1.7 version has been stable for 2 years or more, so this seems a bit surprising.

Please list the smallest modlist (with exact mod names and version numbers) which causes the error, or provide a log.

**Comment by EzerArch** [comment]
> Small testing mod list

I have exact all these mods except Vampirism. In fact, I have 53 mods installed, no issues.

I also get stack overflow if I mess too much with the Options > Video Settings.

**Comment by ROMVoid95** [comment]
I created an Instance with those exact mods, and i noticed that the startup messages seen in the second picture that say "something has an update". those were not present after 4 worlds in my instance. verify the mods installed

in case those are. As Radfast stated, Logs would be the most helpful. Also could help checking these.
- make sure your GPU drivers are updated (fixed it once before)
- i also remember seeing this back in 1.7.x and if i recall my friend had me download a specific jre version and placed it in my java folder and put that folder path as my Java Executable in the launcher, fixed it as well. Think it was 1.8.0_25 if i remember right.

**Comment by VenomiXeD** [comment]
Alright. My drivers are always updated.

Also. We're running this as a modpack (private) and sure I could tell them to get the specific jre. But I don't think that would work for that long. But try using howling moon with gc.

**Comment by radfast** [comment]
@VenomiXeD :

>    Please list the smallest modlist (with exact mod names and version numbers) which causes the error, or provide a log.

**Comment by VenomiXeD** [comment]
Testing now. I'm using the smallest modlist I possibly can.
![image](https://user-images.githubusercontent.com/32819948/66857070-cae25f00-ef86-11e9-8a57-4b45c717c19e.png)
As you can see. The result is the same. The wing gets displaced. And with more mods or less. It'll cause water to be all glitchy and have visual errors.
![image](https://user-images.githubusercontent.com/32819948/66857112-e188b600-ef86-11e9-99d5-3aeff7b47953.png)
Here's the mod list

Logs:
[fml-client-latest.log](https://github.com/micdoodle8/Galacticraft/files/3730808/fml-client-latest.log)
[latest.log](https://github.com/micdoodle8/Galacticraft/files/3730809/latest.log)


Analyzing this visual bug. I suspect that it tries to render the wing as if it was translated to the normal's player's position and rotation. Or something along those lines.
![image](https://user-images.githubusercontent.com/32819948/66857274-375d5e00-ef87-11e9-8fdf-1087ea0bb224.png)

**Comment by VenomiXeD** [comment]
> > Small testing mod list
> 
> I have all these mods except Vampirism. In fact, I have 53 mods installed, no issues.
> 
> I also get stack overflow if I mess too much with the Options > Video Settings.

Then try with Vampirism in a cheat world and then turn in to a bat. (Use "Cheat a vampire level", Middle click button and then drag mouse towards the bat icon, then press F5)

**Comment by marcus8448** [comment]
I'm confused on what the "smallest" modlist that reproduces this is. The log that you sent has **16** mods loaded, while the screenshot (of the modlist) says there should be less. Could you send the *latest* log that uses the least amount of mods (that still reproduces this issue)?

**Comment by VenomiXeD** [comment]
Oh. Might be because I have "Core mods" in the 1.7.10. such as forge multipart etc

I was unaware. I'll do a fresh forge installation and then just the GC mod and Vampirism. And then upload logs and such. Sounds good?

**Comment by EzerArch** [comment]
> Sounds good?

Yes. Also, does your current installation have Optifine? Just to be sure.

**Comment by VenomiXeD** [comment]
This is false. We do have BetterFPS and FastCraft, But the issue remains even without the mods.

**Comment by VenomiXeD** [comment]
Now continuing this issue. Reporting that the core issue lies on GalacticCraft. Because here's my modlist.
![image](https://user-images.githubusercontent.com/32819948/67642770-6ec8f480-f90f-11e9-93b9-bd97ec1a7731.png)

As you can see. This time I made sure that the 1.7.10 folder is empty:
![image](https://user-images.githubusercontent.com/32819948/67642774-81432e00-f90f-11e9-85a8-8fa5e050f404.png)

Here are some results.
![image](https://user-images.githubusercontent.com/32819948/67642759-548f1680-f90f-11e9-84e2-bc2072cb43d4.png)

Also occurs in normal terrain generation.
![image](https://user-images.githubusercontent.com/32819948/67642784-ba7b9e00-f90f-11e9-825e-6794e5e1f2da.png)


There's also an issue on the github for their code. (make sure to select 1.7.10(!!!))
https://github.com/TeamLapen/Vampirism/issues/229

Logs:
[fml-client-latest.log](https://github.com/micdoodle8/Galacticraft/files/3776503/fml-client-latest.log)
[latest.log](https://github.com/micdoodle8/Galacticraft/files/3776504/latest.log)

As you can read of the latest.log. It just says stackoverflow 1283 @ POST render, so I guess glPopmatrix somewhere buggy thing?

**Comment by radfast** [comment]
Thanks for the further info.

**Comment by marcus8448** [comment]
Well, I was able to reproduce this issue.
Crashed the game, as I was using LWJGL's debug tools (`-Dorg.lwjgl.util.Debug=true`). Sadly, there isn't really any useful info in the crash report: https://paste.ee/p/mcF9U

**Comment by VenomiXeD** [comment]
Interesting to know. But how so. There's no stacktrace of where it actually came from... So that's actually quite weird.

**Comment by VenomiXeD** [comment]
Just to keep this issue a bit on going. Is there any reports on what's doing this? This isn't supposed to stress anyone up. But are there any theories yet? Any thoughts or so? I do suspect this though:

One of GC's classes are transforming the player class. Then when rendered. Vampirism also has a transformer class for rendering (I think). Which I guess makes this issue a thing... But I ain't some Forge expert. I have only skimmed through the source code to see what it is.

**Comment by VenomiXeD** [comment]
So, here we are, got any news for this issue? I don't expect a fix or something, but more of an update on what is known or so.
